repository/config.go

```diff
@@ -0,0 +1,23 @@
+package repository
+
+// Config represent the common function interacting with the repository config storage
+type Config interface {
+	// Store writes a single key/value pair in the config of the repo
+	Store(key string, value string) error
+
+	// ReadAll reads all key/value pair matching the key prefix
+	ReadAll(keyPrefix string) (map[string]string, error)
+
+	// ReadBool read a single boolean value from the config
+	// Return ErrNoConfigEntry or ErrMultipleConfigEntry if
+	// there is zero or more than one entry for this key
+	ReadBool(key string) (bool, error)
```

Comment:
Should we really keep different Read functions in the interface ? we'll have `ReadBool`, `ReadTimestamp` and `ReadString` probably more in the future.
Or maybe have one `Read(key string) (string, error)` and move the conversion utility to another package ?

Reply:
I think it's fine to have that in the interface. It allows the implementations to store the value the way they want if needed. You can still have shared internal helpers to do the conversion if you want to reduce the code duplication.

Reply:
yes, most likely